ocaml/xapi/import_raw_vdi.ml

```diff
@@ -112,17 +112,45 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
               debug "Checking whether localhost can see SR: %s" (Ref.string_of sr);
               if (Importexport.check_sr_availability ~__context sr)
               then localhost_handler rpc session_id vdi req s
-              else
-                let host = Importexport.find_host_for_sr ~__context sr in
-                let address = Db.Host.get_address ~__context ~self:host in
-                return_302_redirect req s address;
-                None
+              else 
+                begin
+                  try 
```

Comment:
I think we could wrap even more code with this try/with block, to catch all errors in the handler. It might be better to wrap the whole import function, to catch any errors that occur in it, by moving this try/with block to wrap lines 161-166 in `let handler ...`.